parsers/dortmund.py

```diff
@@ -0,0 +1,143 @@
+import re
+import datetime
+import urllib.request as rq
+
+from urllib.parse import urlencode as urlencode
+from bs4 import BeautifulSoup
+from utils import Parser
+
+from pyopenmensa.feed import LazyBuilder
+
+# dictionary for categories
+categories = {
+    103: 'Tagesgericht',
+    101: 'Menü 1',
+    102: 'Menü 2',
+    104: 'Vegetarisches Menü',
+    105: 'Aktionsteller',
+    106: 'Aktionsteller Fisch',
+    107: 'Aktionsteller Vegan',
+    108: 'Grillstation',
+    109: 'Selfservice Station',
+    115: 'Restaurant',
+    116: 'Beilage',
+    120: 'Menü 1',
+    121: 'Menü 2',
+    122: 'Menü 3',
+    123: 'Menü 3',
+    124: 'Vegetarisches Menü',
+    125: 'Vegan',
+    127: 'Aktionsteller',
+    128: 'Schnelle Theke',
+    130: 'Sonstige',
+    131: 'Low Carb Teller',
+    135: 'Pastatheke Pastaauswahl',
+    136: 'Pastatheke Toppings',
+    137: 'Pastatheke Saucen',
+}
+
+def getWeekdays(day):
+    currentWeekday = day.weekday()
+    week = []
+    wstart = -currentWeekday
+    wend = 7 - currentWeekday
+    for d in range(wstart, wend):
+        nday = day + datetime.timedelta(days=d)
+        week.append(nday.isoformat())
+    return week
+
+def parse_url(url, today=False):
+    canteen = LazyBuilder()
+
+    canteen.extra_regex = re.compile('\((?P<extra>[0-9a-zA-Z]{1,3}'
+                                     '(?:,[0-9a-zA-Z]{1,3})*)\)', re.UNICODE)
+
+    legend_url = 'https://www.stwdo.de/mensa-co/allgemein/zusatzstoffe/'
+    canteen.legend = parse_legend(legend_url)
+
+    day = datetime.date.today()
+    week = getWeekdays(day)
+
+    for wDay in week:
+        py = {'tx_pamensa_mensa[date]' : wDay}
+        payload = urlencode(py).encode('ascii')
+        data = rq.urlopen(url, payload).read().decode('utf-8')
+        soup = BeautifulSoup(data, 'html.parser')
+        parse_day(canteen, soup, wDay)
+
+    return canteen.toXMLFeed()
+
+def parse_legend(url):
+    data = rq.urlopen(url).read().decode('utf-8')
+    soup = BeautifulSoup(data, 'html.parser')
+
+    table = soup.find('table', { 'class' : 'ce-table' })
+    tbody = soup.find('tbody')
+
+    legend = {}
+
+    for tr in tbody.find_all('tr'):
+        td = tr.find_all('td')
+        key = td[0].text.strip(' ')
```

Comment:
You don't need to specify the characters to remove, strip without parameters will just remove any whitespace.